tools/dds/dds-sniffer/dds-sniffer.cpp

```diff
@@ -333,19 +341,17 @@ void dds_sniffer::save_topic_writer( const eprosima::fastrtps::rtps::WriterDisco
 {
     std::string topic_name = info.info.topicName().c_str();
     _discovered_topics[topic_name].writers.insert( info.info.guid() );
-    save_max_indentation( std::move( topic_name ) );
 }
 
 void dds_sniffer::remove_topic_writer( const eprosima::fastrtps::rtps::WriterDiscoveryInfo & info )
 {
-    auto topic_entry = _discovered_topics[info.info.topicName().c_str()];
-    auto writer_iter = topic_entry.writers.find( info.info.guid() );
-    if( writer_iter != topic_entry.writers.end() )
+    auto topic_entry = _discovered_topics.find(info.info.topicName().c_str());
+    if(topic_entry != _discovered_topics.end() )
     {
-        topic_entry.writers.erase( writer_iter );
-        if( topic_entry.writers.empty() && topic_entry.readers.empty() )
+        topic_entry->second.writers.erase( info.info.guid() );
+        if( topic_entry->second.writers.empty() && topic_entry->second.readers.empty() )
         {
-            _discovered_topics.erase( info.info.topicName().c_str() );
+            _discovered_topics.erase( topic_entry );
```

Comment:
Does any of this need a mutex? Can a reader and writer, for example, be simultaneously added/removed?

Reply:
Additions and removals of entries are done from a single listener thread, but main thread and listener threads can access entities concurrently. Will add mutex to synchronize.